Src/WitsmlExplorer.Api/Workers/DownloadLogDataWorker.cs

```diff
@@ -114,4 +149,361 @@ private string GetReportBody(ICollection<Dictionary<string, LogDataValue>> repor
+    {
+        writer.WriteLine("#");
+        writer.WriteLine(
+            "#-----------------------------------------------------------");
+        int i = 0;
+        var line = new StringBuilder();
+        foreach (var curveSpecification in curveSpecifications)
+        {
+            int emptySpaces = maxColumnLenghts[curveSpecification.Mnemonic] -
+                              curveSpecification.Mnemonic.Length -
+                              curveSpecification.Unit.Length - 3;
+            if (i == 0)
+            {
+                line.Append("# ");
+                if (emptySpaces > 2)
+                    line.Append(new string(' ', emptySpaces - 2));
+            }
+            else
+            {
+                if (emptySpaces > 0)
+                    line.Append(new string(' ', emptySpaces));
+            }
+
+            line.Append(curveSpecification.Mnemonic);
+            line.Append(" (");
+            line.Append(curveSpecification.Unit);
+            if (i < curveSpecifications.Count)
+                line.Append(") ");
+            else
+            {
+                line.Append(')');
+            }
+            i++;
+        }
+        writer.WriteLine(line.ToString());
+        writer.WriteLine(
+            "#-----------------------------------------------------------");
+        writer.WriteLine("~A");
+    }
+
+    private void WriteWellInformationSection(StringWriter writer, Well well, int maxColumnLength, int maxDataLenght, LimitValues limitValues)
+    {
+        writer.WriteLine("~WELL INFORMATION BLOCK");
+        CreateHeader(writer, maxColumnLength, maxDataLenght, "#MNEM", ".UNIT", "DATA", "DESCRIPTION OF MNEMONIC");
+        WriteWellParameter(writer, "STRT", limitValues.Unit, limitValues.Start, $"START {limitValues.LogType}", maxColumnLength, maxDataLenght);
+        WriteWellParameter(writer, "STOP", limitValues.Unit, limitValues.Stop, $"STOP {limitValues.LogType}", maxColumnLength, maxDataLenght);
+        WriteWellParameter(writer, "STEP", limitValues.Unit, limitValues.Step, "STEP VALUE", maxColumnLength, maxDataLenght);
+        WriteWellParameter(writer, "SNULL", "", "", "NULL VALUE", maxColumnLength, maxDataLenght);
+        WriteWellParameter(writer, "SCOMP", "", well.Operator, "COMPANY NAME", maxColumnLength, maxDataLenght);
+        WriteWellParameter(writer, "SWELL", "", well.Name, "WELL NAME", maxColumnLength, maxDataLenght);
+        WriteWellParameter(writer, "SFLD", "", "", "FIELD NAME", maxColumnLength, maxDataLenght);
+        WriteWellParameter(writer, "SRIGN", "", "", "RIG NAME", maxColumnLength, maxDataLenght);
+        WriteWellParameter(writer, "SRIGTYP", "", "", "RIG TYPE", maxColumnLength, maxDataLenght);
+        WriteWellParameter(writer, "SSON", "", "", "Service Order Number", maxColumnLength, maxDataLenght);
+        WriteWellParameter(writer, "SSRVC", "", "", "SERVICE COMPANY NAME", maxColumnLength, maxDataLenght);
```

Comment:
Can get this from `logObject.ServiceCompany`.

Reply:
fixed

Reply:
done